File: packages/java/src/main/java/com/slybrowser/WebDriverLaunchSettings.java
package com.slybrowser;

import java.nio.file.Path;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.Collections;
import java.util.List;
import java.util.Map;

public final class WebDriverLaunchSettings {
  public Object profile = Collections.emptyMap();
  public Path tempRoot;
  public Path releaseRoot;
  public Path profileDir;
  public String profileMode;
  public final List<String> browserArguments = new ArrayList<>();
  public final List<String> excludeSwitches = new ArrayList<>(
      Arrays.asList("enable-automation", "enable-unsafe-swiftshader"));
  public boolean headless = true;
  public int viewportWidth = 1920;
  public int viewportHeight = 947;
  public boolean humanize;
  public String humanPreset = "default";
  public Map<String, Number> humanConfig;
  public Integer humanSeed;
  public Object runtimeHandoff;
  public Object driverRuntimeHandoff;
  public boolean allowRuntimeActivationTicket;
  public boolean nativeReady;
  public long nativeReadyTimeoutMs = 15_000;
  public long driverStartTimeoutMs = 15_000;
  public long commandTimeoutMs = 60_000;
}

File: packages/dotnet/src/SlyBrowser/WebDriverLaunchSettings.cs
namespace SlyBrowser;

public sealed class WebDriverLaunchSettings
{
    public object Profile { get; init; } = new { };
    public object? RuntimeHandoff { get; init; }
    public object? DriverRuntimeHandoff { get; init; }
    public bool AllowRuntimeActivationTicket { get; init; }
    public bool NativeReady { get; init; }
    public TimeSpan NativeReadyTimeout { get; init; } = TimeSpan.FromSeconds(15);
    public string? TempRoot { get; init; }
    public string? ReleaseRoot { get; init; }
    public string? ProfileDirectory { get; init; }
    public string? ProfileMode { get; init; }
    public IList<string> BrowserArguments { get; init; } = [];
    public IList<string> ExcludedSwitches { get; init; } = ["enable-automation", "enable-unsafe-swiftshader"];
    public bool Headless { get; init; } = true;
    public int ViewportWidth { get; init; } = 1920;
    public int ViewportHeight { get; init; } = 947;
    public bool Humanize { get; init; }
    public string HumanPreset { get; init; } = "default";
    public IReadOnlyDictionary<string, double>? HumanConfig { get; init; }
    public int? HumanSeed { get; init; }
    public TimeSpan DriverStartTimeout { get; init; } = TimeSpan.FromSeconds(15);
    public TimeSpan CommandTimeout { get; init; } = TimeSpan.FromSeconds(60);
}
